File: basic_ros_example/src/service_advertiser.cpp
#include <ros/ros.h>
#include <basic_ros_example/Adder.h>

bool srvCallback(basic_ros_example::Adder::Request& req, basic_ros_example::Adder::Response& res) {
  res.result = req.val1 + req.val2;
  return true;
}

int main(int argc, char** argv) {
  ros::init(argc, argv, "service_advertiser");
  ros::NodeHandle node;

  ros::ServiceServer srv = node.advertiseService("adder_service", srvCallback);

  ros::spin();
}

File: basic_ros_example/scripts/basic_ros_example/service_advertiser.py
#! /usr/bin/env python3
import rospy
from basic_ros_example.srv import Adder, AdderResponse


class ServiceAdvertiser:
    def __init__(self):
        self.srv = rospy.Service('adder_service', Adder, self.srv_callback)

    def srv_callback(self, req):
        res = AdderResponse()
        res.result = req.val1 + req.val2
        return res


if __name__ == '__main__':
    rospy.init_node('service_advertiser')
    node_instance = ServiceAdvertiser()
    rospy.spin()